apps/voting/contracts/Voting.sol

```diff
@@ -600,48 +622,33 @@ contract Voting is IForwarder, AragonApp {
-        if (voterIndex < delegatedVotersCount - 1) {
-            delegatedVoters[_delegate].addresses[voterIndex] = lastVoter;
+        address[] storage votersList = delegatedVoters[_delegate].addresses;
+        assert(votersList[voterIndex] == _voter);
+
+        uint256 lastVoterIndex = votersList.length - 1;
+        if (voterIndex < lastVoterIndex) {
+            address lastVoter = votersList[lastVoterIndex];
+            votersList[voterIndex] = lastVoter;
             delegates[lastVoter].voterIndex = voterIndex;
         }
-        delegatedVoters[_delegate].addresses.length--;
-        emit ResetDelegate(_voter, _delegate);
+        votersList.length--;
+        delete delegates[_voter];
+        emit UnassignDelegate(_voter, _delegate);
     }
 
     /**
-     * @notice Return the sliced list of delegated voters for `_delegate` and their voting power at the `_blockNumber`
-     * @param _delegate the address of the delegate
-     * @param _offset the number of delegated voters from the start of the list to skip
-     * @param _limit the number of delegated voters to return
-     * @param _blockNumber the block number to get the voting power
-     * @return the array of delegated voters
-     * @return the array of voting power of delegated voters
+     * @dev internal function to get the voting power of the `_voters` at the `_blockNumber`
+     * @param _voters list of voters
+     * @param _blockNumber target block number
      */
-    function _getDelegatedVotersAt(address _delegate, uint256 _offset, uint256 _limit, uint256 _blockNumber) internal view returns (address[] memory votersList, uint256[] memory votingPowerList) {
-        require(_delegate != address(0), ERROR_ZERO_ADDRESS_PASSED);
-        require(_limit > 0, ERROR_INVALID_LIMIT);
-        uint256 delegatedVotersCount = delegatedVoters[_delegate].addresses.length;
-        if (delegatedVotersCount == 0) {
-            return (votersList, votingPowerList);
-        }
-        require(_offset < delegatedVotersCount, ERROR_INVALID_OFFSET);
-
-        uint256 returnCount = _offset.add(_limit) > delegatedVotersCount ? delegatedVotersCount.sub(_offset) : _limit;
-        votersList = new address[](returnCount);
-        votingPowerList = new uint256[](returnCount);
-        address voter;
-        for (uint256 i = 0; i < returnCount; ++i) {
-            voter = delegatedVoters[_delegate].addresses[_offset + i];
-            votersList[i] = voter;
-            votingPowerList[i] = token.balanceOfAt(voter, _blockNumber);
+    function _getVotingPowerMultipleAt(address[] _voters, uint256 _blockNumber) internal view returns (uint256[] memory balances) {
+        uint256 votersCount = _voters.length;
+        balances = new uint256[](votersCount);
+        for (uint256 i = 0; i < votersCount; ++i) {
+            balances[i] = token.balanceOfAt(_voters[i], _blockNumber);
         }
-        return (votersList, votingPowerList);
+        return balances;
```

Comment:
No need to explicitly return `balances`

Reply:
Addressed in f1f6de1606a6eda1999a0e1cc0ea895857101f88.